calico_containers/tests/unit/calicoctl_test.py

```diff
@@ -176,6 +181,708 @@ def test_validate_arguments(self, case, sys_exit_called):
+
+    @patch('sys.exit', autospec=True)
+    @patch('calico_ctl.container.enforce_root', autospec=True)
+    @patch('calico_ctl.container.get_container_info_or_exit', autospec=True)
+    @patch('calico_ctl.container.client', autospec=True)
+    @patch('calico_ctl.container.get_pool_or_exit', autospec=True)
+    @patch('calico_ctl.container.netns', autospec=True)
+    def test_container_add_container_not_running(
+            self, m_netns, m_get_pool_or_exit, m_client,
+            m_get_container_info_or_exit, m_enforce_root, m_sys_exit):
+        """
+        Assert that system exits when a container is not running
+        """
+        # Set up mock object
+        m_client.get_endpoint.side_effect = KeyError
+        m_get_container_info_or_exit.return_value = {
+            'Id': 666,
+            'State': {'Running': 0, 'Pid': 'Pid_info'}
+        }
+
+        # Call method under test
+        container.container_add('container1', '1.1.1.1', 'interface')
+
+        # Assert
+        m_sys_exit.assert_called_once_with(1)
+
+    @patch('sys.exit', autospec=True)
+    @patch('calico_ctl.container.enforce_root', autospec=True)
+    @patch('calico_ctl.container.get_container_info_or_exit', autospec=True)
+    @patch('calico_ctl.container.client', autospec=True)
+    @patch('calico_ctl.container.get_pool_or_exit', autospec=True)
+    @patch('calico_ctl.container.netns', autospec=True)
+    def test_container_add_not_ipv4_configured(
+            self, m_netns, m_get_pool_or_exit, m_client,
+            m_get_container_info_or_exit, m_enforce_root, m_sys_exit):
+        """
+        Assert that system exits when client cannot obtain next hop IPs
+        """
+        # Set up mock objects
+        m_client.get_endpoint.side_effect = KeyError
+        m_client.get_default_next_hops.return_value = {}
+
+        # Call method under test
+        container.container_add('container1', '1.1.1.1', 'interface')
+
+        # Assert
+        m_sys_exit.assert_called_once_with(1)
+
+    @patch('sys.exit', autospec=True)
+    @patch('calico_ctl.container.enforce_root', autospec=True)
+    @patch('calico_ctl.container.get_container_info_or_exit', autospec=True)
+    @patch('calico_ctl.container.client', autospec=True)
+    @patch('calico_ctl.container.get_pool_or_exit', autospec=True)
+    @patch('calico_ctl.container.netns', autospec=True)
+    def test_container_add_ip_previously_assigned(
+            self, m_netns, m_get_pool_or_exit, m_client,
+            m_get_container_info_or_exit, m_enforce_root, m_sys_exit):
+        """
+        Assert that system exits when an ip address is already assigned in pool
+        """
+        # Set up mock object
+        m_client.get_endpoint.side_effect = KeyError
+        m_client.assign_address.return_value = []
+
+        # Call method under test
+        container.container_add('container1', '1.1.1.1', 'interface')
+
+        # Assert
+        m_sys_exit.assert_called_once_with(1)
```

Comment:
Gave this be the assertRaises(SystemExit... like I mentioned above so you can remove the patches and/or test that the patched funcs are not called.